app/routes/index.js

```diff
@@ -5,6 +5,20 @@ export default Ember.Route.extend({
 
   kredits: Ember.inject.service(),
 
+  beforeModel(transition) {
+    const kredits = this.get('kredits');
+
+    if (kredits.get('web3') && kredits.get('web3Provided')) {
```

Comment:
No, because the `web3` property actually checks if there is a user provided web3 when called for the first time. And it's not being called before this, so we have to access it first.